### ISSUE TITLE
Spell checker is not working

### ISSUE THREAD

**Issue by ghost** [florisboard/florisboard#1523]
#### Short description
Spell checker is not working at all.

#### Steps to reproduce
1. I imported 2 dictionaries (English and German) to FlorisBoard (the aff and dic files) -> Florisboard shows them
2. I activated the spell check in the system settings, set Florisboard as default keyboard and as default spell checker
3. When I type incorrect words (Email, SMS, Messenger,...) nothing happens at all

#### Environment information
- FlorisBoard Version: 0.3.13
- Install Source: F-Droid
- Device: Huawei P30
- Android: 10
- EMUI 11

**Comment by NovaViper** [comment]
Can confirm the same issue, also on the same version but my phone's a Samsung S9 running Android 9

**Comment by MythicalNewt** [comment]
![Screenshot_20230816_225516_FlorisBoard](https://github.com/florisboard/florisboard/assets/89124240/a77b0f26-0e84-44e1-a35e-aa65185dae0b)

Can confirm the same on samsung device.
Looks like spellcheck and spelling correction doesn't work.

As seen in the above image typing a word suggestions have a numeral after them.

**Comment by itisafrickinhighlander** [comment]
I'm not sure if this issue is still being tracked, but I remember that I once used Floris' built-in spell checking and it worked. I did some research and testing today and I see that it worked until 4.x release, so I went back to the [v0.3.16](https://github.com/florisboard/florisboard/releases/tag/v0.3.16) version and it does indeed work for me just fine.

Was this first-gen spellchecker really that buggy that it had to be disabled altogether? I saw mentions in the first 4.x release that it might be optionally available again, so any plans to do so? I will honestly stay on 0.3.16 as it otherwise seems to be working just fine for me (Pixel 8A, Android 15).

Last note, I also made sure that in the Android Spell checker settings I chose "Use keyboard spell checker config" as the option for Languages setting
![Image](https://github.com/user-attachments/assets/dfefc26e-86d4-45db-8772-d0c226e68939)

**Comment by patrickgold** [comment]
> Does returning the pre-v4 spellchecker have any blocker?

Yes, licensing wise it is not really compatible with our project's license.

**Comment by Kladki** [comment]
> Yes, licensing wise it is not really compatible with our project's license.

According to https://nuspell.github.io, it's licensed under LGPL-3.0-or-later, which allows it to be used in projects with any license, even proprietary licenses.

Am I missing something here, or was there some misubderstanding of the terms of the license?
